src/zcbor_decode.c

```diff
@@ -204,6 +204,18 @@ bool zcbor_int64_decode(zcbor_state_t *state, int64_t *result)
 }
 
 
+bool zcbor_uint_decode(zcbor_state_t *state, void *result_uint, size_t uint_size)
+{
+	INITIAL_CHECKS_WITH_TYPE(ZCBOR_MAJOR_TYPE_PINT);
```

Comment:
Just curious, why does this differ so much from `int_decode` ? There we don't do the check with type.

Reply:
Actually there is the check, in line 165, it is just inlned and little bit cut down.

Reply:
Exactly. For int_decode, the number can be either positive or negative, which are two different CBOR types. INITIAL_CHECKS_WITH_TYPE() cannot be used to check for two types.